e2e/upgrade_test.go

```diff
@@ -136,17 +139,33 @@ var _ = Context("upgrade", func() {
 			if err != nil {
 				return err
 			}
-			for _, cond := range cluster.Status.Conditions {
-				if cond.Type != mocov1beta2.ConditionHealthy {
-					continue
-				}
-				if cond.Status == corev1.ConditionTrue {
-					return nil
-				}
-				return fmt.Errorf("cluster is not healthy: %s", cond.Status)
+			if cluster.Generation != cluster.Status.ReconcileInfo.Generation {
+				return fmt.Errorf("cluster is not reconciled yet")
 			}
-			return errors.New("no health condition")
+			conditionHealthy := meta.FindStatusCondition(cluster.Status.Conditions, mocov1beta2.ConditionHealthy)
+			if conditionHealthy == nil {
+				return fmt.Errorf("cluster is not healthy")
+			}
+			if conditionHealthy.Status != metav1.ConditionTrue {
+				return fmt.Errorf("cluster is not healthy")
+			}
+			conditionUpToDate := meta.FindStatusCondition(cluster.Status.Conditions, mocov1beta2.ConditionUpToDate)
+			if conditionUpToDate == nil {
+				return fmt.Errorf("statefulset is not updated yet")
+			}
+			if conditionUpToDate.Status != metav1.ConditionTrue {
+				return fmt.Errorf("statefulset is not updated yet")
+			}
+			return nil
 		}).Should(Succeed())
+
+		sts := &appsv1.StatefulSet{}
+		out, err := kubectl(nil, "get", "-n", "upgrade", "sts", "moco-test", "-o", "json")
+		Expect(err).NotTo(HaveOccurred())
+		err = json.Unmarshal(out, sts)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(sts.Status.CurrentRevision).Should(Equal(sts.Status.UpdateRevision))
+		Expect(sts.Generation).Should(Equal(sts.Status.ObservedGeneration))
```

Comment:
I want to check that `UpToDate` is `false` while the StatefulSet is rolling updating.

Reply:
WIP
I'm considering how to check UpToDate while updating.

Reply:
Oh, sorry for the early comment.

Reply:
Sorry for missleading reply. I'm currently working to check `UpToDate` is false while updating in response to your review.